mcstatus/motd/simplifies.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from mcstatus.motd.components import Formatting, MinecraftColor, ParsedMotdComponent, WebColor
+
+
+def get_unused_elements(parsed: Sequence[ParsedMotdComponent]) -> set[int]:
+    """Get indices of all items which should be removed from MOTD as unused.
+
+    This is a wrapper method around several item collection methods, which do so based on several principles.
+    """
+    to_remove: set[int] = set()
+
+    for simplifier in [get_double_items, get_double_colors, get_formatting_before_color, get_empty_text, get_end_non_text]:
+        to_remove.update(simplifier(parsed))
+
+    return to_remove
+
+
+def get_double_items(parsed: Sequence[ParsedMotdComponent]) -> set[int]:
+    """Get indices of all doubled items that can be removed."""
+    to_remove: set[int] = set()
+
+    for index, item in enumerate(parsed):
+        try:
+            next_item = parsed[index + 1]
+        except IndexError:  # Last item (without any next item)
+            break
+
+        if item == next_item:
```

Comment:
We can add `squash_double_strings` in simplifies, but I will try to ~~re~~produce this bug firstly.

Reply:
Actually,
```py
>>> Motd.parse({"text": "123", "extra": [{"text": "123"}]}).simplify().parsed
['123', <Formatting.RESET: 'r'>, '123']
>>> Motd(["123", "123"]).simplify().parsed
['123']
```
But first, is more other kind of bug - simplifies don't simplify enough (will fix it later).

Reply:
Fixed [here](https://github.com/py-mine/mcstatus/compare/0c41416ccd29a421d1afbb72019576950a76a95b..a950e9998c7e7bea0094900b7875933de6c9f696), thanks!

P.S. I'm open for ideas, if you have better idea how to fix it.

Reply:
Yeah, the simplification logic could probably be a lot better, I'm not sure what's the best way to handle this rn. 

However I'd heavily suggest not changing the simplifies in this PR anymore, I'd like to see this merged at least somewhat soon, and this could then be something for another PR to address, for now it works well enough and it fixes some pretty major bugs with current parsing. Also, several people have asked for this recently on discord, so it'd be quite nice to have this merged, instead of having them install from a fork.

Reply:
Maybe split this PR to three parts? The base (this PR, `motd/__init__.py` and `motd/components.py`), simplifies and transformers.

This would help to merge the PR faster, and will help in review.

---

P.S. While writing this message, I noticed that this PR has a `do-not-merge` tag, and so it can't be merged before #306 anyway.

Reply:
The do-not-merge is purely here because of the conflict in naming (description being renamed to motd), which is supposed to be addressed by 306 somehow, I wouldn't consider this PR to actually be blocked though, since while originally, the intention here was to make description hold the actual `Motd` class, right now, it's just calling the `to_minecraft`, method, which is fully backwards compatible, and making motd available as a whole which potentially breaks compatibility can be then handled elsewhere. I don't consider this PR as blocked.

Also, I'm against further splitting this PR, once all of my pending reviews are addressed, unless there's some significant changes to the code, most of it is already reviewed by me and so I'll likely just approve, or only add some minor comments to the newly added code since after this review, so it's pretty much almost ready to be merged.

Though it will still need kevin's review after my approval, which might hold this off for a while longer, but nevertheless, I don't think we should split this, and it shouldn't be that big holdoff.

Reply:
Then... maybe remove the `do-not-merge` tag?